python/pyspark/java_gateway.py

```diff
@@ -201,7 +201,7 @@ def local_connect_and_auth(port, auth_secret):
         af, socktype, proto, _, sa = res
         try:
             sock = socket.socket(af, socktype, proto)
-            sock.settimeout(15)
+            sock.settimeout(int(os.environ.get("SPARK_GATEWAY_CONNECT_TIMEOUT", 15)))
```

Comment:
@gaborgsomogyi, I think this doesn't cover the driver side case (not worker side). This auth is used in driver side too, for example, in `context.py` or `rdd.py` when the data has to be ser/de between driver JVM and Python NVM.